group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1314,8 +1331,12 @@ private void throwIfConsumerGroupHeartbeatRequestIsInvalid(
             if (request.topicPartitions() == null || !request.topicPartitions().isEmpty()) {
                 throw new InvalidRequestException("TopicPartitions must be empty when (re-)joining.");
             }
-            if (request.subscribedTopicNames() == null || request.subscribedTopicNames().isEmpty()) {
-                throw new InvalidRequestException("SubscribedTopicNames must be set in first request.");
+            boolean hasSubscribedTopics =
+                    request.subscribedTopicNames() != null && !request.subscribedTopicNames().isEmpty();
```

Comment:
nit: I wonder if we could have a small helper method for this. We could then reuse it at L1337.

Reply:
they both look very similar indeed, but `subscribedTopicNames` is list<string> and `subscribedTopicRegex` is a string so a reusable helper for the empty check would be a bit tricky, probably making it more complicated I'm afraid? not sure if I missing something